src/app/components/project-search/project-search.component.ts

```diff
@@ -0,0 +1,86 @@
+import { Component, OnInit } from '@angular/core';
+import { headerConfigKeys, urlConstants } from 'src/app/core/constants';
+import { HttpService, ToastService } from 'src/app/core/services';
+import { UtilService } from 'src/app/shared/util.service';
+
+@Component({
+  selector: 'app-project-search',
+  templateUrl: './project-search.component.html',
+  styleUrls: ['./project-search.component.scss'],
+})
+export class ProjectSearchComponent  implements OnInit {
+
+  searchResults: any[] = [];
+  currentPage = 1;
+  isLoading = false;
+  debounceTimer: any;
+  limit = 2;
+  public title: string = 'GLOBAL_SEARCH.TITLE'
+  searchTerm: any = "";
+
+  constructor(
+    private http: HttpService,
+    private toast: ToastService,
+    private utilsService: UtilService
+    ) {}
+  ngOnInit(): void {
+    this.getProjects();
+    this.utilsService.setHeaders({
+      [headerConfigKeys.SHOW_ICON]: true,
+      [headerConfigKeys.SHOW_MENU]: true,
+      [headerConfigKeys.SHOW_SEARCH]: false,
+      [headerConfigKeys.SHOW_NOTIFICATION]: false,
+      [headerConfigKeys.SHOW_PROFILE]: true,
+      })
+  }
+
+  onSearch(event: any) {
+    this.searchTerm = event.target.value.trim();
+    
+    if (this.searchTerm.length >= 3) {
+      this.currentPage = 1;
+
+      // Debounce the API call to avoid frequent requests while typing
+      clearTimeout(this.debounceTimer);
+      this.debounceTimer = setTimeout(() => {
+        this.getProjects();
+      }, 300);
+    }
+  }
+
+  async getProjects() {
+    const dynamicUrl = urlConstants.API_URLS.GLOBAL_SEARCH(this.currentPage, this.limit, this.searchTerm);
+    const config = {
+      url: dynamicUrl,
+    };
+
+    await this.http.setHeader();
+    this.http.get(config).subscribe((data) => {
+      this.isLoading = false;
+      if (data) {
+        if(data.result == 0){
```

Comment:
what is this condition?

Reply:
it generates the toast in infinite scroll and informs the user that no data is available